### PR TITLE
apache-arrow apache-arrow-glib 17.0.0

### PR DESCRIPTION
**Pull request by lukeshingles**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/a/apache-arrow-glib.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
   desc "GLib bindings for Apache Arrow"
   homepage "https://arrow.apache.org/"
-  url "https://www.apache.org/dyn/closer.lua?path=arrow/arrow-16.1.0/apache-arrow-16.1.0.tar.gz"
-  mirror "https://archive.apache.org/dist/arrow/arrow-16.1.0/apache-arrow-16.1.0.tar.gz"
-  sha256 "c9e60c7e87e59383d21b20dc874b17153729ee153264af6d21654b7dff2c60d7"
+  url "https://www.apache.org/dyn/closer.lua?path=arrow/arrow-17.0.0/apache-arrow-17.0.0.tar.gz"
+  mirror "https://archive.apache.org/dist/arrow/arrow-17.0.0/apache-arrow-17.0.0.tar.gz"
+  sha256 "9d280d8042e7cf526f8c28d170d93bfab65e50f94569f6a790982a878d8d898d"
   license "Apache-2.0"
   revision 1
```
Comment (raulcd) [inline Formula/a/apache-arrow-glib.rb:8]:
```suggestion
```

**Review by raulcd — COMMENTED** [review]
shouldn't we also `bump-revisions` for dependencies?
```
Formula/g/gdal.rb
Formula/v/vineyard.rb
Formula/d/dvc.rb
```

**Comment by lukeshingles** [comment]
I’m afraid I don’t really understand what is failing there. If the CI jobs are flaky, could someone with access retry just the failed jobs? Otherwise, any suggestions on how to fix this?

**Review by branchv — APPROVED** [review]
@lukeshingles thank you!

**Pull request merged**